stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ApplicationDescriptor.java

```diff
@@ -58,106 +51,36 @@ protected Object onGetChildAt(Application element, int index) {
-
-    public abstract List<Activity> getActivitiesList();
-  }
+    public ElementContext(Application element) {
+      mElement = element;
+    }
 
-  private final class ElementContextPreICS extends ElementContext {
-    @Override
     public void hook(Application element) {
+      mActivityTracker.registerListener(mListener);
     }
 
-    @Override
     public void unhook() {
+      mActivityTracker.unregisterListener(mListener);
     }
 
-    @Override
     public List<Activity> getActivitiesList() {
-      return ApplicationUtil.getAllActivities();
+      return mActivityTracker.getActivitiesView();
     }
-  }
 
-  @TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
-  private final class ElementContextICS extends ElementContext {
-    private Application mElement;
-    private Application.ActivityLifecycleCallbacks mCallbacks;
-    private List<Activity> mActivities;
-
-    @Override
-    public void hook(Application element) {
-      mElement = Util.throwIfNull(element);
-
-      // TODO: tree diffing will remove the need to even worry about installing this callback,
-      //       which will then allow ~realtime updates for pre-ICS. for now, pre-ICS will just
-      //       have to close and re-open the inspector whenever the activity changes.
-      //       (this does assume that the hack below for getAllActivitiesHack() works on pre-ICS)
-      mCallbacks = new Application.ActivityLifecycleCallbacks() {
-        @Override
-        public void onActivityCreated(Activity activity, Bundle savedInstanceState) {
-          mActivities.add(0, activity);
-          getHost().onChildInserted(mElement, null, activity);
-        }
-
-        @Override
-        public void onActivityStarted(Activity activity) {
-        }
-
-        @Override
-        public void onActivityResumed(Activity activity) {
-        }
-
-        @Override
-        public void onActivityPaused(Activity activity) {
-        }
-
-        @Override
-        public void onActivityStopped(Activity activity) {
-        }
-
-        @Override
-        public void onActivitySaveInstanceState(Activity activity, Bundle outState) {
-        }
-
-        @Override
-        public void onActivityDestroyed(Activity activity) {
-          mActivities.remove(activity);
-          getHost().onChildRemoved(mElement, activity);
-        }
-      };
-
-      mElement.registerActivityLifecycleCallbacks(mCallbacks);
-
-      mActivities = new ArrayList<Activity>(ApplicationUtil.getAllActivities());
-    }
-
-    @Override
-    public void unhook() {
-      if (mElement != null) {
-        if (mCallbacks != null) {
-          mElement.unregisterActivityLifecycleCallbacks(mCallbacks);
-          mCallbacks = null;
-        }
-
-        mElement = null;
+    private final ActivityTracker.Listener mListener = new ActivityTracker.Listener() {
+      @Override
+      public void onActivityAdded(Activity activity) {
+        getHost().onChildInserted(mElement, null, activity);
```

Comment:
Hmm, you're right. It was. *scratches head* Okay, leave it alone but file an issue about it and assign it to me.